src/CachePlugin.php

```diff
@@ -293,16 +293,16 @@ private function getETag(CacheItemInterface $cacheItem)
             return;
         }
 
-        if (is_array($data['etag'])) {
-            foreach ($data['etag'] as $etag) {
-                if (!empty($etag)) {
-                    return $etag;
-                }
-            }
+        if (!is_array($data['etag'])) {
+            return $data['etag'];
+        }
 
-            return;
+        foreach ($data['etag'] as $etag) {
+            if (!empty($etag)) {
+                return $etag;
+            }
         }
 
-        return $data['etag'];
+        return;
```

Comment:
actually, it's fine because php will internally make it null whenever you try to use it, you just can't go the other way around if you wanted void but returned null